lib/game/base_game.dart

```diff
@@ -0,0 +1,192 @@
+import 'dart:math' as math;
+import 'dart:ui';
+
+import 'package:flame/components/composed_component.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart' hide WidgetBuilder;
+import 'package:flutter/foundation.dart';
+import 'package:ordered_set/comparing.dart';
+import 'package:ordered_set/ordered_set.dart';
+
+import '../components/component.dart';
+import '../components/mixins/has_game_ref.dart';
+import '../components/mixins/tapable.dart';
+import '../position.dart';
+import '../gestures.dart';
+import 'game.dart';
+
+/// This is a more complete and opinionated implementation of Game.
+///
+/// It still needs to be subclasses to add your game logic, but the [update], [render] and [resize] methods have default implementations.
+/// This is the recommended structure to use for most games.
+/// It is based on the Component system.
+class BaseGame extends Game with TapDetector {
+  /// The list of components to be updated and rendered by the base game.
+  OrderedSet<Component> components =
+      OrderedSet(Comparing.on((c) => c.priority()));
```

Comment:
this should fit in a single line I think, no?

Reply:
i let `flutter format` decide that, don't you use it?